cmd/minikube/cmd/docker-env.go

```diff
@@ -172,6 +182,17 @@ var dockerEnvCmd = &cobra.Command{
 			}
 		}
 
+		out, err := tryDockerConnectivity("docker", ec)
+		if err != nil { // docker might be up but been loaded with wrong certs/config
+			glog.Warningf("couldn't connect to docker inside minikube. output: %s error: %v", string(out), err)
+			// to fix issues like this #8185
+			glog.Infof("will try to restart dockerd service...")
+			mustRestartDocker(cname, co.CP.Runner)
+			// TODO #8241: use kverify to wait for apisefver instead
+			// waiting for the basics like api-server to come up
+			time.Sleep(time.Second * 3)
```

Comment:
Remove this sleep until it can be proved that this is helpful.

Reply:
done, but I am sure ppl will run into this, specially people who use this as part of another tool chain unvisible to the user